src/pkg/caendr/caendr/services/user.py

```diff
@@ -7,15 +7,17 @@
   'admin': 'Admin'
 }
 
+# TODO: Replace with Entity.query_ds call? Should this return as User Entity class?
```

Comment:
It looks like there are two places where this function is used:

- On the registration page, to check if this email is already associated with an account.  Since all we care about here is whether the email is in use or not, I think it's alright to query by email.
  ```
  def validate_email(form, field):
    existing_user = get_local_user_by_email(field.data)
    if len(existing_user):
      raise ValidationError("Email already exists")
  ```
  Code [here](https://github.com/AndersenLab/CAENDR/blob/feature/multiple-species-etl-ops/src/modules/site-v2/base/forms/forms.py#L113).

- To reset a user's password.  I'm less sure what to do here -- the existing code just takes the first result, and uses that username to generate a one time token.
  ```
  email = request.form.get("email")
  users = get_local_user_by_email(email)

  if not users or not users[0]:
    logger.error(f"Failed to retrieve LOCAL user for email: {email}")
    flash('Unable to reset your credentials at this time. Please try again later.', 'error')
    return redirect('/')
    
  user = users[0]
  ```
  Code [here](https://github.com/AndersenLab/CAENDR/blob/feature/multiple-species-etl-ops/src/modules/site-v2/base/views/user.py#L60).

How should we handle the password reset page?